src/test/gitSecrets.test.ts

```diff
@@ -0,0 +1,133 @@
+    }
+
+    /** git-secrets patterns that will not cause a failure during the scan */
+    function setAllowListPatterns(gitSecrets: string) {
+        const allowListPatterns: string[] = ['^node_modules/', '^\\.[a-zA-Z]', '^dist/']
+
+        allowListPatterns.forEach(pattern => {
+            // Returns non-zero exit code if pattern already exists
+            runCmd([gitSecrets, '--add', '--allowed', pattern], { cwd: toolkitProjectDir, throws: false })
+        })
+    }
+
+    function setDenyListPatterns(gitSecrets: string) {
+        const denyListPatterns: string[] = []
+
+        denyListPatterns.forEach(pattern => {
+            // Returns non-zero exit code if pattern already exists
+            runCmd([gitSecrets, '--add', pattern], { cwd: toolkitProjectDir, throws: false })
+        })
+    }
+
+    function deleteFileIfExists(filePath: string) {
+        if (existsSync(filePath)) {
+            unlinkSync(filePath)
+        }
+    }
+
+    before(function () {
+        if (platform() === 'win32') {
+            this.skip()
+        }
+
+        toolkitProjectDir = path.join(getProjectDir(), '..', '..')
+        testFixturesPath = path.join(toolkitProjectDir, 'src', 'testFixtures', 'resources')
+        mkdirSync(testFixturesPath, { recursive: true })
+
+        gitSecrets = path.join(testFixturesPath, 'git-secrets')
+        setupGitSecretsExecutable(gitSecrets)
+
+        accessKeyFilePath = path.join(testFixturesPath, 'fileWithAccessKey.ts')
+        deleteFileIfExists(accessKeyFilePath)
+
+        // Register all patterns with `git-secrets`
+        runCmd([gitSecrets, '--register-aws'], { cwd: toolkitProjectDir })
+        setDenyListPatterns(gitSecrets)
+        setAllowListPatterns(gitSecrets)
+    })
+
+    afterEach(function () {
+        deleteFileIfExists(accessKeyFilePath)
+    })
+
+    it('ensures no git secrets are found', function () {
+        const result = runCmd([gitSecrets, '--scan', '--no-index'], { cwd: toolkitProjectDir })
+        assert.strictEqual(result.status, 0, `Failure output: ${result.stderr.toString()}`)
+    })
+
+    it('sanity check it finds secrets', function () {
+        // Create file in project that has secret key value.
+        // Need to build access key string incrementally to not trigger git-secrets.
+        const keyValue = 'yAki21XLhAIBiKvyaxr4p/ltr8OxkZTHISISFAKE'
+        const mySecretAccessKey = `const x = { "aws_secret_access_key": "${keyValue}" }`
+        const fileContent = `
+/*!
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+${mySecretAccessKey}
+`.trim()
+        writeFileSync(accessKeyFilePath, fileContent)
+
+        const result = runCmd([gitSecrets, '--scan', '--no-index'], { cwd: toolkitProjectDir, throws: false })
+
+        assert.strictEqual(result.status, 1)
+    })
+})
+
+function runCmd(args: string[], options?: SpawnSyncOptions & { throws?: boolean }) {
```

Comment:
does using our existing `src/shared/utilities/childProcess.ts` avoid the need for this?